// flex-layouts/lib/column/Column.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _elementum = require("elementum");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = _elementum.StyleSheet.createClassMap({
  "self": "FLColumnColumn",
  "fill": "FLColumnColumn--Fill",
  "wrap": "FLColumnColumn--Wrap",
  "align=start": "FLColumnColumn--AlignStart",
  "align=center": "FLColumnColumn--AlignCenter",
  "align=end": "FLColumnColumn--AlignEnd",
  "justify=start": "FLColumnColumn--JustifyStart",
  "justify=center": "FLColumnColumn--JustifyCenter",
  "justify=end": "FLColumnColumn--JustifyEnd",
  "justify=spaceAround": "FLColumnColumn--JustifySpaceAround",
  "justify=spaceBetween": "FLColumnColumn--JustifySpaceBetween",
  "overflow=hidden": "FLColumnColumn--OverflowHidden",
  "overflowX=hidden": "FLColumnColumn--OverflowXHidden",
  "overflowY=hidden": "FLColumnColumn--OverflowYHidden"
});

var Column = function Column(_ref) {
  var children = _ref.children,
      fill = _ref.fill,
      wrap = _ref.wrap,
      align = _ref.align,
      justify = _ref.justify,
      overflow = _ref.overflow,
      overflowX = _ref.overflowX,
      overflowY = _ref.overflowY;
  return _react.default.createElement("div", {
    className: styles({
      fill: fill,
      wrap: wrap,
      align: align,
      justify: justify,
      overflow: overflow,
      overflowX: overflowX,
      overflowY: overflowY
    })
  }, children);
};

var _default = Column;
exports.default = _default;
