// flex-layouts/lib/layout/Layout.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _elementum = require("elementum");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = _elementum.StyleSheet.createClassMap({
  "self": "FLLayoutLayout",
  "relative": "FLLayoutLayout--Relative",
  "align=start": "FLLayoutLayout--AlignStart",
  "align=center": "FLLayoutLayout--AlignCenter",
  "align=end": "FLLayoutLayout--AlignEnd",
  "justify=start": "FLLayoutLayout--JustifyStart",
  "justify=center": "FLLayoutLayout--JustifyCenter",
  "justify=end": "FLLayoutLayout--JustifyEnd",
  "overflow=hidden": "FLLayoutLayout--OverflowHidden",
  "overflowX=hidden": "FLLayoutLayout--OverflowXHidden",
  "overflowY=hidden": "FLLayoutLayout--OverflowYHidden"
});

var Layout = function Layout(_ref) {
  var children = _ref.children,
      _ref$grow = _ref.grow,
      grow = _ref$grow === void 0 ? 0 : _ref$grow,
      _ref$shrink = _ref.shrink,
      shrink = _ref$shrink === void 0 ? 0 : _ref$shrink,
      _ref$basis = _ref.basis,
      basis = _ref$basis === void 0 ? 'auto' : _ref$basis,
      align = _ref.align,
      justify = _ref.justify,
      relative = _ref.relative,
      overflow = _ref.overflow,
      overflowX = _ref.overflowX,
      overflowY = _ref.overflowY;
  return _react.default.createElement("div", {
    className: styles({
      align: align,
      justify: justify,
      overflow: overflow,
      relative: relative,
      overflowX: overflowX,
      overflowY: overflowY
    }),
    style: {
      flex: "".concat(grow, " ").concat(shrink, " ").concat(basis)
    }
  }, children);
};

var _default = Layout;
exports.default = _default;
