// flex-layouts/lib/row/Row.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _elementum = require("elementum");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = _elementum.StyleSheet.createClassMap({
  "self": "FLRowRow",
  "fill": "FLRowRow--Fill",
  "wrap": "FLRowRow--Wrap",
  "align=start": "FLRowRow--AlignStart",
  "align=center": "FLRowRow--AlignCenter",
  "align=end": "FLRowRow--AlignEnd",
  "justify=start": "FLRowRow--JustifyStart",
  "justify=center": "FLRowRow--JustifyCenter",
  "justify=end": "FLRowRow--JustifyEnd",
  "justify=spaceAround": "FLRowRow--JustifySpaceAround",
  "justify=spaceBetween": "FLRowRow--JustifySpaceBetween",
  "overflow=hidden": "FLRowRow--OverflowHidden",
  "overflowX=hidden": "FLRowRow--OverflowXHidden",
  "overflowY=hidden": "FLRowRow--OverflowYHidden"
});

var Row = function Row(_ref) {
  var children = _ref.children,
      fill = _ref.fill,
      wrap = _ref.wrap,
      align = _ref.align,
      justify = _ref.justify,
      overflow = _ref.overflow,
      overflowX = _ref.overflowX,
      overflowY = _ref.overflowY;
  return _react.default.createElement("div", {
    className: styles({
      fill: fill,
      wrap: wrap,
      align: align,
      justify: justify,
      overflow: overflow,
      overflowX: overflowX,
      overflowY: overflowY
    })
  }, children);
};

var _default = Row;
exports.default = _default;
